src/modules/workflowsettings/Elsa.WorkflowSettings/Services/WorkflowSettingsManager.cs

```diff
@@ -32,5 +34,23 @@ public async Task<WorkflowSetting> LoadSettingAsync(string workflowBlueprintId,
 
             return value;
         }
+
+        public async Task<IEnumerable<WorkflowSetting>> LoadSettingsAsync(string workflowBlueprintId, CancellationToken cancellationToken = default, IOrderBy<WorkflowSetting>? orderBy = default, IPaging? paging = default)
+        {
+            var providers = _workflowSettingsProviders;
+
+            var value = new List<WorkflowSetting>();
+
+            foreach (var provider in providers)
+            {
+                var providerValue = await provider.GetWorkflowSettingsAsync(workflowBlueprintId, cancellationToken, orderBy, paging);
+                if (providerValue != null)
+                {
+                    value = providerValue.ToList();
```

Comment:
Instead of overriding the result of the last evaluated provider, you should replace existing values.
It should work similar to the way configuration works in .NET: you can have appsettings.json and environment variables.

Depending on which provider was registered last, takes precedence (in case both providers define the same setting key).

So what you might do instead is build a dictionary of key/setting pairs and simply set the setting into the dicitonary by key.

If the entry didn't exist before, it will be added. Otherwise it will be overridden.

Example:

```csharp
var providers = _workflowSettingsProviders;

            var dictionary= new Dictionary<string, WorkflowSetting>();

            foreach (var provider in providers.OrderByDescending(x => x.Priority) // Let's introduce a priority so we can control what providers take precedence.
            {
                // Since you return a list, it should never be null; always return a list even if empty.
                var settings = await provider.GetWorkflowSettingsAsync(workflowBlueprintId, cancellationToken: cancellationToken);

                foreach(var setting in settings)
                    dictionary[setting.Name] = setting;
            }

            return dictionary.Values;
```

Reply:
Introduced priority property and refactored method according to example.